backend/lib/pix-service/service.js

```diff
@@ -0,0 +1,43 @@
+import { Value } from '@sinclair/typebox/value'
+import { PIECE_INDEXER_URL } from '../config.js'
+import { getMinerPeerId } from '../rpc-service/service.js'
+import { PixResponse } from './data-types.js'
+import pRetry from 'p-retry'
+
+/**
+ * @param {string} providerId
+ * @param {string} pieceCid
+ * @returns {Promise<string>}
+ */
+export const getDealPayloadCid = async (providerId, pieceCid) => {
+  const url = PIECE_INDEXER_URL + '/sample/' + providerId + '/' + pieceCid
+  try {
+    const response = await pRetry(async () => await fetch(url, {
+      method: 'GET',
+      headers: { 'content-type': 'application/json' }
+    }), { retries: 5 })
+    const json = await response.json()
+    try {
+      const parsedPixResponse = Value.Parse(PixResponse, json)
+      if (parsedPixResponse.samples.length === 0) {
+        throw new Error('No samples found in the response.')
+      }
```

Comment:
This is expected, it will happen often (in ~60% of calls).

I propose that `getDealPayloadCid()` should return `undefined` when this happens.

That would also help with addressing the comment above:

```
export const indexPieces = async (makeRpcRequest, makePixRequest, pgPool, queryLimit) => {
  // TODO: handle payloads which cannot be retrieved from the piece CID indexer
```